api/src/main/java/org/openmrs/module/attachments/AttachmentsContext.java

```diff
@@ -196,6 +210,43 @@ protected Boolean getBooleanByGlobalProperty(String globalPropertyName) {
 		return BooleanUtils.toBoolean(globalProperty);
 	}
 	
+	protected Concept getConceptByGlobalProperty(String globalPropertyName) {
+		String globalProperty = administrationService.getGlobalProperty(globalPropertyName);
+		Concept concept = conceptService.getConceptByUuid(globalProperty);
+		if (concept == null) {
+			throw new IllegalStateException("Configuration required: " + globalPropertyName);
+		}
+		return concept;
+	}
+	
+	protected EncounterType getEncounterTypeByGlobalProperty(String globalPropertyName, boolean required) {
+		String globalProperty = administrationService.getGlobalProperty(globalPropertyName);
+		EncounterType encounterType = encounterService.getEncounterTypeByUuid(globalProperty);
+		if (required && encounterType == null) {
+			throw new IllegalStateException("Configuration required: " + globalPropertyName);
+		}
+		return encounterType;
+	}
+	
+	protected Integer getIntegerByGlobalProperty(String globalPropertyName) {
+		String globalProperty = getGlobalProperty(globalPropertyName, true);
+		try {
+			return Integer.valueOf(globalProperty);
+		}
+		catch (Exception e) {
+			throw new IllegalStateException("Global property " + globalPropertyName + " value of " + globalProperty
+			        + " is not parsable as an Integer");
+		}
```

Comment:
Always best to attach the underlying exception to the new exception where possible:

```suggestion
		catch (Exception e) {
			throw new APIException("Global property " + globalPropertyName + " with value " + globalProperty
			        + " is not parsable as an Integer", e);
		}
```